### PR TITLE
Formatting for query and plan on query metrics page (#1593)

### PR DESCRIPTION
**Pull request by kevinrr888**
Currently have the formatMetrics(List<QueryMetric> metrics) method in JexlFormattedStringBuildingVisitor. Can be moved to a more appropriate class if desired.

### CODE REVIEW

**Comment by kevinrr888** [comment]
Currently have the formatMetrics(List<QueryMetric> metrics) method in JexlFormattedStringBuildingVisitor. Can be moved to a more appropriate class if desired.

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/JexlFormattedStringBuildingVisitor.java [hunk 1]
```diff
@@ -305,6 +310,58 @@ public Object visit(ASTReferenceExpression node, Object data) {
...
+                updatedMetric.setPlan(buildQuery(planNode));
+            } catch (ParseException e) {
+                log.error("Could not parse JEXL AST after performing transformations to run the query", e);
+                
+                for (String line : PrintingVisitor.formattedQueryStringList(planNode)) {
+                    log.error(line);
+                }
+                log.error("");
+                
+                QueryException qe = new QueryException(DatawaveErrorCode.QUERY_EXECUTION_ERROR, e);
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/jexl/visitors/JexlFormattedStringBuildingVisitor.java]: If we cannot parse the query, then lets leave the original query in the metric.  This should not cause an exception.

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/JexlFormattedStringBuildingVisitor.java [hunk 2]
```diff
@@ -305,6 +310,58 @@ public Object visit(ASTReferenceExpression node, Object data) {
...
+                    updatedMetric.setQuery(buildQuery(queryNode));
+                } catch (ParseException e) {
+                    log.error("Could not parse JEXL AST after performing transformations to run the query", e);
+                    
+                    for (String line : PrintingVisitor.formattedQueryStringList(queryNode)) {
+                        log.error(line);
+                    }
+                    log.error("");
+                    
+                    QueryException qe = new QueryException(DatawaveErrorCode.QUERY_EXECUTION_ERROR, e);
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/jexl/visitors/JexlFormattedStringBuildingVisitor.java]: If we fail to parse the query, then simply leave the original query in the metric.  We should not be throwing an exception here.

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/JexlFormattedStringBuildingVisitor.java [hunk 3]
```diff
@@ -305,6 +310,58 @@ public Object visit(ASTReferenceExpression node, Object data) {
...
+            String plan = updatedMetric.getPlan();
+            // If it is a JEXL query, set the query to be formatted
+            if (isJexlQuery(metric.getParameters())) {
+                try {
+                    queryNode = JexlASTHelper.parseJexlQuery(query);
+                    updatedMetric.setQuery(buildQuery(queryNode));
+                } catch (ParseException e) {
+                    log.error("Could not parse JEXL AST after performing transformations to run the query", e);
+                    
+                    for (String line : PrintingVisitor.formattedQueryStringList(queryNode)) {
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/jexl/visitors/JexlFormattedStringBuildingVisitor.java]: Lets change this to:
```
if (log.isTraceEnabled()) {
  log.trace(PrintingVisitor.formattedQueryString(queryNode));
}
```

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/JexlFormattedStringBuildingVisitor.java [hunk 4]
```diff
@@ -305,6 +310,58 @@ public Object visit(ASTReferenceExpression node, Object data) {
...
+                }
+            }
+            // Format the plan (plan will always be a JEXL query)
+            try {
+                planNode = JexlASTHelper.parseJexlQuery(plan);
+                updatedMetric.setPlan(buildQuery(planNode));
+            } catch (ParseException e) {
+                log.error("Could not parse JEXL AST after performing transformations to run the query", e);
+                
+                for (String line : PrintingVisitor.formattedQueryStringList(planNode)) {
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/jexl/visitors/JexlFormattedStringBuildingVisitor.java]: Lets change this for loop to:
```
if (log.isTraceEnabled()) {
  log.trace(PrintingVisitor.formattedQueryString(queryNode));
}
```

**Review by ivakegg — CHANGES_REQUESTED** [review]

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/JexlFormattedStringBuildingVisitor.java [hunk 5]
```diff
@@ -305,6 +310,52 @@ public Object visit(ASTReferenceExpression node, Object data) {
...
+                try {
+                    queryNode = JexlASTHelper.parseJexlQuery(query);
+                    updatedMetric.setQuery(buildQuery(queryNode));
+                } catch (ParseException e) {
+                    log.error("Could not parse JEXL AST after performing transformations to run the query", e);
+                    
+                    if (log.isTraceEnabled()) {
+                        log.trace(PrintingVisitor.formattedQueryString(queryNode));
+                    }
+                    log.error("");
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/jexl/visitors/JexlFormattedStringBuildingVisitor.java]: remove this line

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/JexlFormattedStringBuildingVisitor.java [hunk 6]
```diff
@@ -305,6 +310,52 @@ public Object visit(ASTReferenceExpression node, Object data) {
...
+            try {
+                planNode = JexlASTHelper.parseJexlQuery(plan);
+                updatedMetric.setPlan(buildQuery(planNode));
+            } catch (ParseException e) {
+                log.error("Could not parse JEXL AST after performing transformations to run the query", e);
+                
+                if (log.isTraceEnabled()) {
+                    log.trace(PrintingVisitor.formattedQueryString(planNode));
+                }
+                log.error("");
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/jexl/visitors/JexlFormattedStringBuildingVisitor.java]: remove this line

**Review by ivakegg — CHANGES_REQUESTED** [review]

**Review by ivakegg — APPROVED** [review]

**Pull request merged**